### PR TITLE
Azure Resources Page

### PR DESCRIPTION
**Pull request by anishprasad01**
Still in-progress, but wanted to get some feedback now. SDK, Tools, and Tutorial links that I think will be useful are already done. Helpful Tidbits needs work.

EDIT: Added some security info in here.

### CODE REVIEW

File: _docs/azure_resources.md [hunk 1]
```diff
@@ -6,4 +6,59 @@ order: 3
...
 ---
 
-# Azure Resources, SDK Links, Tutorials, and Helpful Tidbits
+# Azure Resources, SDK Links, Tutorials, Helpful Tidbits, and Permissions Management
+
+## SDKs and Tools
+### SDKs
+Azure offers two types of SDKs. Unified SDKs are built on a "common core" and are recommended for all new projects. Standard SDKs are avilable for older projects or languages unsupported by Unified SDKs.
+
+- [Unified SDKs](https://azure.github.io/azure-sdk/releases/latest/index.html#java-packages) available for: .NET, Java, Python, TypeScript/JavaScript
```
Comment (Chris-Johnston) [inline _docs/azure_resources.md]:
```suggestion
- [Unified SDKs](https://azure.github.io/azure-sdk/releases/latest/index.html) available for: .NET, Java, Python, TypeScript/JavaScript
```
might as well link using the top of the page

File: _docs/azure_resources.md [hunk 2]
```diff
@@ -6,4 +6,59 @@ order: 3
...
+- [Standard Java SDKs](https://azure.github.io/azure-sdk/releases/latest/all/java.html) 
+- [Standard Python SDKs](https://azure.github.io/azure-sdk/releases/latest/all/python.html) 
+- [Standard TypeScript/JavaScript SDKs](https://azure.github.io/azure-sdk/releases/latest/all/js.html) 
+- [Standard PHP SDKs](https://packagist.org/packages/microsoft/) 
+- [Standard Go SDKs](https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-install) 
+- [Azure SDKs Page](https://azure.microsoft.com/en-us/downloads/)
+
+### Tools
+- [Visual Studio 2019](https://visualstudio.microsoft.com/) has excellent integrations with Azure services, especially when using C#/.NET.
+- [Azure Visual Studio](https://azure.microsoft.com/en-us/products/visual-studio/)
```
Comment (Chris-Johnston) [inline _docs/azure_resources.md]: seems like a redundant link?

File: _docs/azure_resources.md [hunk 3]
```diff
@@ -6,4 +6,59 @@ order: 3
...
+- [Microsoft Learn - Azure](https://docs.microsoft.com/en-us/learn/azure/)
+- [Azure Documentation](https://docs.microsoft.com/en-us/azure/?product=featured)
+- [Azure 15-Minute Tutorials](https://tutorials.visualstudio.com/)
+- [Getting Started with Visual Studio](https://visualstudio.microsoft.com/vs/getting-started/)
+- [Developing Azure Functions with Visual Studio](https://docs.microsoft.com/en-us/azure/azure-functions/functions-develop-vs)
+- [Deploying Azure Applications with VSCode](https://code.visualstudio.com/docs/azure/deployment)
+- [The Developer's Guide to Azure](https://azure.microsoft.com/en-us/campaigns/developer-guide/)
+
+## Helpful Tidbits
+- Most resources on Azure require supporting resources to be provisioned alongside the main offering. For example, provisioning a VM will also require a Storage Account, Virtual Networking, etc. These resources will remain provisioned *even if you delete the main resource.* Remember to delete **all** resources when you are done using them to avoid charges.
```
Comment (Chris-Johnston) [inline _docs/azure_resources.md]: To add to this, this is why "resource groups" are a thing. Under a subscription you can create many resource groups that represent individual projects. When it comes time to delete resources, you can just delete the whole resource group.

File: _docs/azure_resources.md [hunk 4]
```diff
@@ -6,4 +6,59 @@ order: 3
...
+
+- The protocols that Azure supports for communicating with its services differs from that of AWS or GCP. Refer to the documentation for information specific to each service.
+
+## Security and Permissions Management
+### Role-Based Access Control (RBAC)
+Access to Azure resources is managed through RBAC. RBAC allows you to define how to grant access. It can be done based on resource type, subscriptions, resource groups, or the specific resources themselves. Before continuing, see the [Azure documentation page on RBAC](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview). RBAC uses the concept of a **role assignment** to assign permissions. Role assignments have three parts.
+
+- **Security Principals** are objects that represent the entity that must be granted permissions. There are four tupes of supportal prinipals: users, groups, service prinicpals, and managed identities. See the documentation page for more info on what each type of security principal is for.
```
Comment (Chris-Johnston) [inline _docs/azure_resources.md]:
```suggestion
- **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the documentation page for more info on what each type of security principal is for.
```

File: _docs/feelgood.md [hunk 5]
```diff
@@ -177,7 +177,7 @@ By default, creating a Lambda function from scratch, adding SNS code, and pressi
...
+- **Schedule expression** is where you define the time interval that the function should be triggered at. FeelGood uses cron syntax identical to the syntax used to schedule recurring jobs in native Unix systems; a good reference on cron syntax we found useful is available [here](https://www.adminschoice.com/crontab-quick-reference). Cron syntax uses UTC time. In FeelGood, the trigger uses the cron syntax `cron(0 18 * * ? *)`, which means every day at 10 AM PST/11 AM PDT. We recommend looking up a converter from UTC to your timezone. Saving and enabling this trigger will run the function at the specified time. AWS has a good explainer on creating CloudWatch Events Rules [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html). AWS also has a good explainer on how to [setup CloudWatch triggers for Lambda using cron syntax](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
```
Comment (Chris-Johnston) [inline _docs/feelgood.md]:
```suggestion
- **Schedule expression** is where you define the time interval that the function should be triggered at. FeelGood uses cron syntax identical to the syntax used to schedule recurring jobs in native Unix systems; a good reference on cron syntax we found useful is available [here](https://www.adminschoice.com/crontab-quick-reference). Cron syntax uses UTC time. In FeelGood, the trigger uses the cron syntax `cron(0 18 * * ? *)`, which means every day at 10 AM PST/11 AM PDT. We recommend looking up a converter from UTC to your timezone. Saving and enabling this trigger will run the function at the specified time. AWS has a good explainer on creating CloudWatch Events Rules [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html). AWS also has a good document on how to [setup CloudWatch triggers for Lambda using cron syntax](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
```

**Review by Chris-Johnston — CHANGES_REQUESTED** [review]
Looks good so far, just a few comments.

One thing I would include would be a link to docs.microsoft.com for all thing Azure/.NET/etc.

File: _docs/azure_resources.md (hunk 1 above)
Comment (anishprasad01) [reply _docs/azure_resources.md]: Good catch, thanks

File: _docs/azure_resources.md (hunk 2 above)
Comment (anishprasad01) [reply _docs/azure_resources.md]: Yeah, definitely

File: _docs/azure_resources.md [hunk 6]
```diff
@@ -6,4 +6,63 @@ order: 3
...
+
+### Take Note of Supported Protocols
+The protocols that Azure supports for communicating with its services differs from that of AWS or GCP. Depending on the application you wish to build, the supported protocols may affect your architecture. Refer to the documentation for information specific to each service.
+
+## Security and Permissions Management
+### Role-Based Access Control (RBAC)
+Access to Azure resources is managed through RBAC. RBAC allows you to define how to grant access. It can be done based on resource type, subscriptions, resource groups, or the specific resources themselves. Before continuing, see the [Azure documentation page on RBAC](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview). RBAC uses the concept of a **role assignment** to assign permissions. Role assignments have three parts.
+
+- **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the documentation page for more info on what each type of security principal is for.
```
Comment (etcadinfinitum) [inline _docs/azure_resources.md]: Add embedded link to docs page for good measure
```suggestion
- **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the [documentation page](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview#security-principal) for more info on what each type of security principal is for.
```

**Review by etcadinfinitum — APPROVED** [review]
LGTM!

**Review by Chris-Johnston — APPROVED** [review]
:shipit:

**Pull request merged**